### PR TITLE
chore: moved recorders to `lib/metrics/recorders`

### LINKED ISSUE #2656: Move middleware, db operation and parsed-statement metric functions to `lib/metrics/recorders`
### Description
There are 3 metrics recorders that do no live in `lib/metrics/recorders`: `_recordOperationMetrics`(lib/shim/datastore-shim.js` and  `middlewareMetricRecorder`(lib/shim/webframework-shim/middleware.js`, and parts of `recordMetrics`(lib/db/parsed-statement.js)

### PR DESCRIPTION
**Pull request by svetlanabrennan**
## Description

Moved 3 metric recorder functions to `lib/metrics/recorders`.

## How to Test

Run unit tests

## Related Issues

Fixes #2656

### CODE REVIEW

File: lib/metrics/recorders/middleware-metric-recorder.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Creates a recorder for middleware metrics.
+ *
+ * @private
+ * @param {object} _shim instance of shim
+ * @param {string} metricName name of metric
+ * @returns {Function} recorder for middleware
+ */
+function makeMiddlewareRecorder(_shim, metricName) {
```
[the remark this answers is not in the document]
Comment (svetlanabrennan) [reply lib/metrics/recorders/middleware-metric-recorder.js:16]: I file name or the function name as well?

[the remark this answers is not in the document]
Comment (bizob2828) [reply lib/metrics/recorders/middleware-metric-recorder.js]: function is fine

File: lib/metrics/recorders/database.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const { DB, ALL } = require('../names')
+const { DESTINATIONS } = require('../../config/attribute-filter')
+
+/**
+ * @this ParsedStatement
+ * @param {TraceSegment}  segment - The segment being recorded.
+ * @param {string}        [scope] - The scope of the segment.
+ */
+
+function _recordMetrics(segment, scope) {
```
Comment (bizob2828) [inline lib/metrics/recorders/database.js:17]: nit but this doesn't have to be `_recordMetrics`. could just be `recordDbMetrics`

File: lib/db/parsed-statement.js [hunk 3]
```diff
@@ -22,53 +21,10 @@ function ParsedStatement(type, operation, collection, raw) {
 }
 
 ParsedStatement.prototype.recordMetrics = function recordMetrics(segment, scope) {
```
Comment (bizob2828) [inline lib/db/parsed-statement.js:23]: I would remove this method and tweak how it's being used in `shim/datastore-shim`

**Review by bizob2828 — CHANGES_REQUESTED** [review]

**Comment by svetlanabrennan** [comment]
@bizob2828 I addressed your comments and pushed up my changes.

File: lib/shim/datastore-shim.js [hunk 4]
```diff
@@ -608,59 +610,7 @@ function _getSpec({ spec, shim, fn, fnName, args }) {
  */
 function _recordQueryMetrics(parsed, segment, scope) {
```
Comment (bizob2828) [inline lib/shim/datastore-shim.js:611]: this function doesn't make sense. I would update where this is getting called to 

```js
segDesc.recorder = recordQueryMetrics.bind(parsed)
```

and then remove this function

File: lib/shim/datastore-shim.js [hunk 5]
```diff
@@ -15,7 +15,9 @@ const Shim = require('./shim')
 const urltils = require('../util/urltils')
 const util = require('util')
 const specs = require('./specs')
+const recordOperationMetrics = require('../../lib/metrics/recorders/database-operation')
 const { DatastoreParameters } = specs.params
+const recordMetrics = require('../../lib/metrics/recorders/database')
```
Comment (bizob2828) [inline lib/shim/datastore-shim.js:20]:
```suggestion
const recordQueryMetrics = require('../../lib/metrics/recorders/database')
```

File: test/unit/parsed-statement.test.js [hunk 6]
```diff
@@ -13,6 +13,7 @@ const { match } = require('../lib/custom-assertions')
 
 const Transaction = require('../../lib/transaction')
```
Comment (bizob2828) [inline test/unit/parsed-statement.test.js:14]: these tests are asserting the new `database` metrics recorder. parsed statement is nothing more than a class now. I would move these tests to `test/unit/metrics-recroders/

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: test/unit/parsed-statement.test.js (hunk 6 above)
Comment (svetlanabrennan) [reply test/unit/parsed-statement.test.js:14]: Got it. I'll move them.

File: lib/shim/datastore-shim.js (hunk 4 above)
Comment (svetlanabrennan) [reply lib/shim/datastore-shim.js:611]: Got it

**Review by bizob2828 — APPROVED** [review]
LGTM

**Pull request merged**